internal/host/host.go

```diff
@@ -148,21 +148,28 @@ func NewManager(log logrus.FieldLogger, db *gorm.DB, eventsHandler events.Handle
 	}
 }
 
-func (m *Manager) RegisterHost(ctx context.Context, h *models.Host) error {
+func (m *Manager) RegisterHost(ctx context.Context, h *models.Host, db *gorm.DB) error {
 	var host models.Host
-	err := m.db.First(&host, "id = ? and cluster_id = ?", *h.ID, h.ClusterID).Error
+	err := db.First(&host, "id = ? and cluster_id = ?", *h.ID, h.ClusterID).Error
 	if err != nil && !gorm.IsRecordNotFoundError(err) {
 		return err
 	}
 
 	pHost := &host
 	if err != nil && gorm.IsRecordNotFoundError(err) {
+		if err := db.Unscoped().Delete(&host, "id = ? and cluster_id = ?", *h.ID, h.ClusterID).Error; err != nil {
```

Comment:
please add a comment  to why it is done and that it will not fail if host does not exist